cdap-app-templates/cdap-etl/hydrator-spark-core/src/main/java/co/cask/cdap/etl/spark/batch/BatchSparkPipelineDriver.java

```diff
@@ -84,10 +93,21 @@ public void run(DatasetContext context) throws Exception {
                                              BatchPhaseSpec.class);
 
     try (InputStream is = new FileInputStream(sec.getLocalizationContext().getLocalFile("HydratorSpark.config"))) {
-      sourceFactory = SparkBatchSourceFactory.deserialize(is);
-      sinkFactory = SparkBatchSinkFactory.deserialize(is);
-      DataInputStream dataInputStream = new DataInputStream(is);
-      stagePartitions = GSON.fromJson(dataInputStream.readUTF(), MAP_TYPE);
+      BufferedReader reader = new BufferedReader(new InputStreamReader(is, Charsets.UTF_8));
+      String object = reader.readLine();
+      SparkBatchSourceSinkFactoryInfo sparkBatchSourceSinkFactoryInfo = GSON.fromJson(object,
+                                                                                      SparkBatchSourceSinkFactoryInfo
+                                                                                        .class);
```

Comment:
This is quite an awkward newline placement. Maybe better if you just shorten the variable name.